test/gss/test.sh

```diff
@@ -0,0 +1,223 @@
+#! /bin/sh
```

Comment:
I think the tests should be rewritten to the python testing framework. But before doing that lets first focus on the new code itself.